.github/workflows/build-and-test.yml

```diff
@@ -289,15 +289,15 @@ jobs:
         if: startsWith( matrix.go-version, '1.23' ) != true
```

Comment:
Related to the comment above, what do we want to do with this part specifically? Currently we're running tests in both go versions, is that the intended behavior? Should we be testing only with go 1.23? go 1.24?

Reply:
We should be testing both versions, that looks a bit weird yep https://github.com/open-telemetry/opentelemetry-collector?tab=readme-ov-file#compatibility

Reply:
Ok, I'll just remove that conditional then... Just creates confusion